server/src/main/java/org/opensearch/cluster/routing/RecoverySource.java

```diff
@@ -274,6 +287,11 @@ public SnapshotRecoverySource(String restoreUUID, Snapshot snapshot, Version ver
             } else {
                 index = new IndexId(in.readString(), IndexMetadata.INDEX_UUID_NA_VALUE);
             }
+            if (FeatureFlags.isEnabled(FeatureFlags.SEARCHABLE_SNAPSHOT) && in.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Comment:
this `FeatureFlags.isEnabled(FeatureFlags.SEARCHABLE_SNAPSHOT) && in.getVersion().onOrAfter(Version.V_3_0_0)` is repeated many times in this PR, can we extract this into one method probably a static one ?

Reply:
Just an FYI to be aware of, you'll want to also change that if this is backported to latest 2.x.